Source/DataSources/PointVisualizer.js

```diff
@@ -204,29 +291,76 @@ define([
                     items.set(entity.id, new EntityData(entity));
                 }
             } else {
-                returnPointPrimitive(items.get(entity.id), unusedIndexes);
+                returnPrimitive(items.get(entity.id), unusedPointIndexes, unusedBillboardIndexes);
                 items.remove(entity.id);
             }
         }
 
         for (i = removed.length - 1; i > -1; i--) {
             entity = removed[i];
-            returnPointPrimitive(items.get(entity.id), unusedIndexes);
+            returnPrimitive(items.get(entity.id), unusedPointIndexes, unusedBillboardIndexes);
             items.remove(entity.id);
         }
     };
 
-    function returnPointPrimitive(item, unusedIndexes) {
+    function returnPrimitive(item, unusedPointIndexes, unusedBillboardIndexes) {
         if (defined(item)) {
             var pointPrimitive = item.pointPrimitive;
             if (defined(pointPrimitive)) {
                 item.pointPrimitive = undefined;
                 pointPrimitive.id = undefined;
                 pointPrimitive.show = false;
-                unusedIndexes.push(pointPrimitive._index);
+                unusedPointIndexes.push(pointPrimitive._index);
+            }
```

Comment:
Shouldn't there be a return above this line.  We can never have a `pointPrimitive` and a `billboard` at the same time, right?